pac4j-core/src/main/java/org/pac4j/core/profile/service/InMemoryProfileService.java

```diff
@@ -0,0 +1,101 @@
+package org.pac4j.core.profile.service;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.definition.CommonProfileDefinition;
+import org.pac4j.core.util.CommonHelper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * In-memory profile service.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+
+public class InMemoryProfileService<U extends CommonProfile> extends AbstractProfileService<U>  {
+
+	public Map<String,Map<String,Object>> profiles;
+	//public AuthenticatorProfileCreator<UsernamePasswordCredentials, U> creator;
+	public Class<U> typeArgumentClass;
+
+	public InMemoryProfileService(Class<U> typeArgumentClass) {
+		this(new HashMap<String,Map<String,Object>>(), typeArgumentClass);
+	}
+
+	public InMemoryProfileService(Map<String,Map<String,Object>> profiles, Class<U> typeArgumentClass) {
+		this.profiles = profiles;
+		this.typeArgumentClass = typeArgumentClass;
+	}
+
+	@Override
+	protected void internalInit(final WebContext context) {
+		CommonHelper.assertNotNull("passwordEncoder", getPasswordEncoder());
+		defaultProfileDefinition(new CommonProfileDefinition<U>(x -> {
+			try {
+				return typeArgumentClass.newInstance();
+			} catch (InstantiationException | IllegalAccessException e) {
+				throw new TechnicalException("Unable to instanciate "+typeArgumentClass+", replace with 'CommonProfile' in simple cases");
+			}
+		}));
+		super.internalInit(context);
+	}
+
+	@Override
+	protected void insert(Map<String, Object> attributes) {
+		String id = (String) attributes.get(getIdAttribute());
+		logger.debug("Inserting doc id: {} with attributes: {}", id, attributes);
+		profiles.put(id, attributes);
+	}
+
+	@Override
+	protected void update(Map<String, Object> attributes) {
+		String id = (String) attributes.get(getIdAttribute());
+		logger.debug("Updating id: {} with attributes: {}", id, attributes);
+		profiles.put(id, attributes);
+	}
+
+	@Override
+	protected void deleteById(String id) {
+		logger.debug("Delete id: {}", id);
+		profiles.remove(id);
+	}
+
+	private Map<String, Object> populateAttributes(final Map<String, Object> rowAttributes, final List<String> names) {
+		final Map<String, Object> newAttributes = new HashMap<>();
```

Comment:
I tried

```java
	private Map<String, Object> populateAttributes(final Map<String, Object> rowAttributes, final List<String> names) {
        return rowAttributes.entrySet().stream()
                .filter(p -> names == null || names.contains(p.getKey()))
                .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue()));
	}
```

but it raises a `NullPointerException` due to [this](https://stackoverflow.com/questions/24630963/java-8-nullpointerexception-in-collectors-tomap) very weird "feature", and working around the feature makes the code quite cumbersome...